pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
Do you think we ought to prevent the new RS from scaling down in case it is saturated and a new RS was adopted by its owning Deployment? If so, I can revert this PR back to its previous state where I was merely fixing IsSaturated.